plugins/destination/neo4j/client/write.go

```diff
@@ -5,32 +5,32 @@ import (
 	"strings"
 
 	"github.com/apache/arrow/go/v13/arrow"
-	"github.com/cloudquery/plugin-sdk/v2/schema"
+	"github.com/cloudquery/plugin-sdk/v3/schema"
 	"github.com/neo4j/neo4j-go-driver/v5/neo4j"
 )
 
-func (c *Client) WriteTableBatch(ctx context.Context, table *arrow.Schema, records []arrow.Record) error {
+func (c *Client) WriteTableBatch(ctx context.Context, table *schema.Table, records []arrow.Record) error {
 	session := c.LoggedSession(ctx, neo4j.SessionConfig{})
 	defer session.Close(ctx)
-	tableName := schema.TableName(table)
+
 	rows := make([]map[string]any, 0)
 	for _, record := range records {
 		rows = append(rows, transformValues(record)...)
 	}
 	var sb strings.Builder
 	sb.WriteString("UNWIND $rows AS row MERGE (t:")
-	sb.WriteString(tableName)
+	sb.WriteString(table.Name)
 	sb.WriteString(" {")
-	pks := schema.PrimaryKeyIndices(table)
+	pks := table.PrimaryKeysIndexes()
```

Comment:
Since it's all column names and nothing to do with indexes, can do it simpler: https://github.com/cloudquery/cloudquery/blob/main/plugins/destination/gremlin/client/write.go#L35-L41